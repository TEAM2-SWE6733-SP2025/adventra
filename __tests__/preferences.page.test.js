import React from "react";
import { render, screen } from "@testing-library/react";
import PreferencesPage from "../src/app/preferences/page";
import "@testing-library/jest-dom";
import { SessionProvider } from "next-auth/react";

describe("ProfilePage Component", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  test("renders the Preferences Page", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const navbarElement = screen.getByRole("navigation");
    expect(navbarElement).toBeInTheDocument();
  });
});

describe("Preferences Page components", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  it("renders the Preferences heading", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const heading = screen.getByRole("heading", {
      name: /Preferences/i,
    });

    expect(heading).toBeInTheDocument();
  });
});

describe("Preferences Page gender filter", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  it("renders the Preferences genders filter", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const femaleBtn = screen.getByRole("button", {
      name: /Female/i,
    });
    const everyoneBtn = screen.getByRole("button", {
      name: /Everyone/i,
    });

    expect(femaleBtn).toBeInTheDocument();
    expect(everyoneBtn).toBeInTheDocument();
  });
});

describe("Preferences Page distance filter", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  it("renders the Preferences distance filter", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const distanceHeading = screen.getByText("Distance:");
    expect(distanceHeading).toBeInTheDocument();

    const sliders = screen.getAllByRole("slider");
    const distanceSlider = sliders.find(
      (slider) => slider.getAttribute("max") === "1000",
    );

    expect(distanceSlider).toBeInTheDocument();
  });
});

describe("Preferences Page age range filter", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  it("renders the Preferences age range filter", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const ageRangeHeading = screen.getByText("Age:");
    expect(ageRangeHeading).toBeInTheDocument();

    const rangeSliderContainer = screen.getByTestId("element");
    expect(rangeSliderContainer).toBeInTheDocument();

    const rangeSliderRange = rangeSliderContainer.querySelector(
      ".range-slider__range",
    );
    expect(rangeSliderRange).toBeInTheDocument();
  });
});

describe("Preferences Page submit Buttons", () => {
  const mockSession = {
    user: {
      name: "Test User",
      email: "test@example.com",
      image: "https://example.com/avatar.png",
    },
    expires: "2025-12-31T23:59:59.999Z",
  };

  it("renders the Preferences submit Buttons", () => {
    render(
      <SessionProvider session={mockSession}>
        <PreferencesPage />
      </SessionProvider>,
    );

    const appleBtn = screen.getByText("Apply");
    expect(appleBtn).toBeInTheDocument();

    const cancelBtn = screen.getByText("Cancel");
    expect(cancelBtn).toBeInTheDocument();
  });
});
