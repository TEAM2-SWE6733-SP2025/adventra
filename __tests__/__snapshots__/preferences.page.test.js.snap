// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preferences Page submit Buttons matches the snapshot 1`] = `
<DocumentFragment>
  <header
    class="block w-full"
  >
    <nav
      class="bg-white dark:bg-black border-b border-gray-200 dark:border-gray-700 w-full"
    >
      <div
        class="max-w-7xl mx-auto px-6 sm:px-8 lg:px-10"
      >
        <div
          class="flex justify-between items-center h-16"
        >
          <div
            class="flex-shrink-0 flex items-center space-x-2"
          >
            <a
              class="lg:px-5"
              href="/preferences"
            >
              <svg
                class="w-5 h-5 text-yellow-500 hover:text-yellow-700"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M495.9 166.6c3.2 8.7 .5 18.4-6.4 24.6l-43.3 39.4c1.1 8.3 1.7 16.8 1.7 25.4s-.6 17.1-1.7 25.4l43.3 39.4c6.9 6.2 9.6 15.9 6.4 24.6c-4.4 11.9-9.7 23.3-15.8 34.3l-4.7 8.1c-6.6 11-14 21.4-22.1 31.2c-5.9 7.2-15.7 9.6-24.5 6.8l-55.7-17.7c-13.4 10.3-28.2 18.9-44 25.4l-12.5 57.1c-2 9.1-9 16.3-18.2 17.8c-13.8 2.3-28 3.5-42.5 3.5s-28.7-1.2-42.5-3.5c-9.2-1.5-16.2-8.7-18.2-17.8l-12.5-57.1c-15.8-6.5-30.6-15.1-44-25.4L83.1 425.9c-8.8 2.8-18.6 .3-24.5-6.8c-8.1-9.8-15.5-20.2-22.1-31.2l-4.7-8.1c-6.1-11-11.4-22.4-15.8-34.3c-3.2-8.7-.5-18.4 6.4-24.6l43.3-39.4C64.6 273.1 64 264.6 64 256s.6-17.1 1.7-25.4L22.4 191.2c-6.9-6.2-9.6-15.9-6.4-24.6c4.4-11.9 9.7-23.3 15.8-34.3l4.7-8.1c6.6-11 14-21.4 22.1-31.2c5.9-7.2 15.7-9.6 24.5-6.8l55.7 17.7c13.4-10.3 28.2-18.9 44-25.4l12.5-57.1c2-9.1 9-16.3 18.2-17.8C227.3 1.2 241.5 0 256 0s28.7 1.2 42.5 3.5c9.2 1.5 16.2 8.7 18.2 17.8l12.5 57.1c15.8 6.5 30.6 15.1 44 25.4l55.7-17.7c8.8-2.8 18.6-.3 24.5 6.8c8.1 9.8 15.5 20.2 22.1 31.2l4.7 8.1c6.1 11 11.4 22.4 15.8 34.3zM256 336a80 80 0 1 0 0-160 80 80 0 1 0 0 160z"
                />
              </svg>
            </a>
            <a
              class="text-2xl font-bold"
              href="/"
            >
              Adventra
            </a>
          </div>
          <button
            class="md:hidden text-gray-500 dark:text-gray-300 hover:text-gray-700 dark:hover:text-gray-100 cursor-pointer"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16m-7 6h7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="hidden md:flex space-x-12 items-center"
          >
            <a
              href="/"
            >
              Home
            </a>
            <a
              href="/about"
            >
              About
            </a>
            <a
              href="/profile"
            >
              Profile
            </a>
             
            <div
              class="flex items-center space-x-4"
            >
              <img
                alt="User Profile"
                class="w-8 h-8 rounded-full"
                src="https://example.com/avatar.png"
              />
              <button
                class="border border-yellow-500 text-yellow-500 px-4 py-2 rounded-lg hover:bg-yellow-900 cursor-pointer"
              >
                Logout
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="md:hidden overflow-hidden transition-all duration-300 ease-in-out max-h-0 opacity-0 bg-white dark:bg-black border-t border-gray-200 dark:border-gray-700"
      >
        <div
          class="px-4 py-3 flex flex-col gap-4"
        >
          <a
            href="/"
          >
            Home
          </a>
          <a
            href="/about"
          >
            About
          </a>
          <a
            href="/profile"
          >
            Profile
          </a>
           
          <div
            class="flex items-center space-x-4"
          >
            <img
              alt="User Profile"
              class="w-8 h-8 rounded-full"
              src="https://example.com/avatar.png"
            />
            <button
              class="border border-yellow-500 text-yellow-500 px-4 py-2 rounded-lg hover:bg-yellow-900 cursor-pointer"
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="min-h-screen flex items-center justify-center p-8"
  >
    <div
      class="max-w-4xl w-full"
    >
      <h1
        class="text-4xl font-extrabold text-yellow-500 mb-8"
      >
        Preferences
      </h1>
      <p
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        Please edit these preferences for your getting better matches.
      </p>
      <div
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        Iâ€™m interested in...
        <div
          class="flex items-center justify-center mb-12"
        >
          <div
            class="relative w-80 h-10 bg-gray-200 rounded-full"
          >
            <div
              class="absolute top-0 left-0 h-full w-1/3 bg-yellow-500 rounded-full transition-transform duration-300 translate-x-[200%]"
            />
            <div
              class="absolute inset-0 flex justify-between items-center px-4"
            >
              <button
                class="text-gray-600 font-medium focus:outline-none px-5 w-1/3 pl-0"
              >
                Male
              </button>
              <button
                class="text-gray-600 font-medium focus:outline-none px-5 w-1/3 "
              >
                Female
              </button>
              <button
                class="text-gray-600 font-medium focus:outline-none px-5 w-1/3 "
              >
                Everyone
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        <div
          class="text-lg dark:text-white text-gray-600 mb-2"
        >
          Distance:
        </div>
        <div
          class="flex flex-col items-center mt-4 border border-solid border-amber-400 rounded-lg p-8 mb-12"
        >
          <span
            class="mt-2 text-gray-600 dark:text-white"
          >
            Selected Distance: 999 Miles
          </span>
          <input
            class="w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer"
            max="1000"
            min="0"
            type="range"
            value="999"
          />
        </div>
      </div>
      <div
        class="text-lg dark:text-white text-gray-600 mb-12"
      >
        <div
          class="text-lg dark:text-white text-gray-600 mb-2"
        >
          Age:
        </div>
        <div
          class="flex flex-col items-center mt-4 border border-solid border-amber-400 rounded-lg p-8 mb-12"
        >
          <span
            class="mt-2 text-gray-600 dark:text-white"
          >
            Selected Range: 18 - 90 Years
          </span>
          <div
            class="range-slider w-full h-2 bg-amber-200 background rounded-lg appearance-none cursor-pointer accent-amber-500"
            data-testid="element"
          >
            <div
              aria-disabled="false"
              aria-valuemax="100"
              aria-valuemin="18"
              aria-valuenow="18"
              aria-valuetext="18"
              class="range-slider__thumb"
              data-lower="true"
              role="slider"
              style="left: calc(0% + NaNpx);"
              tabindex="0"
            />
            <div
              aria-disabled="false"
              aria-valuemax="100"
              aria-valuemin="18"
              aria-valuenow="90"
              aria-valuetext="90"
              class="range-slider__thumb"
              data-upper="true"
              role="slider"
              style="left: calc(87.8048780487805% + NaNpx);"
              tabindex="0"
            />
            <div
              class="range-slider__range"
            />
          </div>
        </div>
      </div>
      <div
        class="mt-12 flex gap-6"
      >
        <button
          class="px-6 py-3 rounded-xl font-semibold transition-all hover:scale-105 bg-yellow-500 text-black hover:bg-yellow-600"
        >
          Apply
        </button>
        <button
          class="px-6 py-3 rounded-xl font-semibold transition-all hover:scale-105 border border-yellow-500 text-yellow-500 hover:bg-yellow-900"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
